lib/multiplexer/multiplexer.go

```diff
@@ -414,75 +433,102 @@ func (m *Mux) detect(conn net.Conn) (*Conn, error) {
 
 		switch proto {
 		case ProtoProxy:
-			if !m.EnableExternalProxyProtocol {
+			newPROXYLine, err := ReadProxyLine(reader)
+			if err != nil {
+				return nil, trace.Wrap(err, invalidProxyLineError)
+			}
+
+			if m.PROXYProtocolMode == PROXYProtocolOff {
 				return nil, trace.BadParameter(proxyProtocolDisabledError)
 			}
-			// We allow only one unsigned proxy line
-			if proxyLine != nil {
-				return nil, trace.BadParameter(duplicateProxyLineError)
+
+			unsignedPROXYLineReceived = true
+
+			if m.PROXYProtocolMode == PROXYProtocolUnspecified {
+				m.Logger.WithFields(log.Fields{
+					"direct_src_addr": conn.RemoteAddr(),
+					"direct_dst_addr": conn.LocalAddr(),
+					"proxy_src_addr:": newPROXYLine.Source.String(),
+					"proxy_dst_addr:": newPROXYLine.Destination.String(),
+				}).Error(unexpectedPROXYLine)
+				newPROXYLine.Source.Port = 0
 			}
-			proxyLine, err = ReadProxyLine(reader)
-			if err != nil {
-				return nil, trace.Wrap(err, invalidProxyLineError)
+
+			if proxyLine != nil {
+				if proxyLine.IsVerified {
```

Comment:
Should this case ever happen?

Reply:
Yeah why shouldn't an error be returned here instead? A signed PROXY header followed by an unsigned PROXY header is allowed?

Reply:
I think it shouldn't, but I wasn't sure if some combination of load balancers and proxy peering didn't lead to it. And if it would happen that would be correct way to handle it, so didn't return an error to lower probability of unnecessary failures. But thinking again now, I couldn't come up when it would possibly happen so changing it to return an error.